engine/src/flutter/lib/web_ui/test/engine/text_editing_test.dart

```diff
@@ -1943,6 +1943,126 @@ Future<void> testMain() async {
       hideKeyboard();
     });
 
+    test('Supports composition changes with shift arrow selection in Japanese IME', () {
+      final HybridTextEditing testTextEditing = HybridTextEditing();
+      final GlobalTextEditingStrategySpy editingStrategy = GlobalTextEditingStrategySpy(
+        testTextEditing,
+      );
```

Comment:
There's already an instance of `HybridTextEditing` being created for each test in `setUp`.

Also, I don't think you need a `GlobalTextEditingStrategySpy`. A simple `DefaultTextEditingStrategy` should work.
```suggestion
      final DefaultTextEditingStrategy editingStrategy = DefaultTextEditingStrategy(textEditing);
```